test-cypress/integration/explore/visual.graph.spec.js

```diff
@@ -382,393 +228,352 @@ describe('Visual graph screen validation', function () {
+
+        openVisualGraphSettings();
+        // Set "vin:Chardonnay" as a preferred type
+        getPreferredTypesField().clear().type('vin:Chardonnay');
+        // Select "Show preferred types only"
+        showPreferredTypes(true);
+
+        saveSettings();
+
+        // Verify that there are a total of 6 ( 5 children plus one parent nodes ) are connected to the DRY node
+        getNodes().should('have.length', 6).each(($el) => {
+            // Exclude parent node
+            if ($el.text() !== 'Dry') {
+                expect($el.text()).to.contain('Chardonnay');
+            }
+        });
     });
 
-    it('Test ignored types', function () {
-        // Search for "http://www.w3.org/TR/2003/PR-owl-guide-20031209/wine#Dry"
-        searchForResource('http://www.w3.org/TR/2003/PR-owl-guide-20031209/wine#Dry');
+    it('Test ignored types', () => {
+        typeInSearchField('http://www.w3.org/TR/2003/PR-owl-guide-20031209/wine#Dry');
 
         // Pick a type that is displayed in the diagram for example "vin:Zinfandel"
-        cy.get('g.nodes-container > g.node-wrapper > foreignObject > div.node-label-body > div')
-            .should('contain', 'Zinfandel');
-
-        cy.get('i.icon-settings.icon-2x').click()
-            .then(() => {
-                // Go to Settings and set "vin:Zinfandel" as an ignored type
-                cy.get('.active > :nth-child(3) > :nth-child(2) > .ng-isolate-scope > .host > .tags > .ng-pristine')
-                    .clear().type('vin:Zinfandel').wait(200);
-
-                // Set the connections limit to 10
-                cy.get('div.filter-sidepanel.tab-content.ng-scope > div > div.form-group.mb-2 > div.form-inline > input[type="number"]')
-                    .clear().type('10');
-
-                cy.get('button[type="submit"]').click().wait(1000)
-                    .then(() => {
-                        // Verify that "vin:Zinfandel" has been removed from the diagram
-                        cy.get('g.nodes-container > g.node-wrapper > foreignObject > div.node-label-body > div')
-                            .should('not.contain', 'Zinfandel');
-                    });
-            });
-
-        // Return previous state of the graph
-        cy.get('i.icon-settings.icon-2x').click()
-            .then(() => {
-                cy.get('.active > :nth-child(3) > :nth-child(2) > .ng-isolate-scope > .host > .tags > .ng-pristine')
-                    .clear();
-
-                cy.get('div.filter-sidepanel.tab-content.ng-scope > div > div.form-group.mb-2 > div.form-inline > input[type="number"]')
-                    .clear().type('20');
-
-                cy.get('button[type="submit"]').click();
-            });
+        getNodes().should('contain', 'Zinfandel');
+
+        openVisualGraphSettings();
+        // Go to Settings and set "vin:Zinfandel" as an ignored type
+        getIgnoredTypesField()
+            .clear().type('vin:Zinfandel').wait(200);
```

Comment:
Why is the wait needed

Reply:
Well, there were too much explicit waits and I just couldn't manage to understand the purpose of all of them so I left some intact. Anyway, I'll remove this one.